### ISSUE TITLE
Adjust TrialBatch to align with new TrialData design

### ISSUE THREAD

**Issue by hmd101** [flatironinstitute/psyphy#138]
## Background

`trial_placement/` provides `GridPlacement` and `SobolPlacement`, both with a `propose(posterior, batch_size) -> TrialBatch` interface. Neither is tested. `TrialBatch` currently stores stimuli as `list[tuple[Any, Any]]` — a hardcoded two-stimulus format that predates the generalised `TrialData` layout introduced in the data-object refactor.

Two additional known issues in `SobolPlacement`: the zero reference vector is hardcoded rather than passed as a parameter, and the class has no tests.

## What this issue covers

**1. Redesign `TrialBatch`**

Change the internal representation from `list[tuple[Any, Any]]` to `list[np.ndarray]` each of shape `(K, d)`:

`# current
stimuli = [((0.0, 0.0), (0.3, -0.5)), ...]   # list of 2-tuples

# proposed
stimuli = [np.array([[0.0, 0.0], [0.3, -0.5]]), ...]  # list of (K, d) arrays`

This aligns `TrialBatch` with `TrialData.stimuli` and lets `ResponseData.add_batch` consume it without conversion — `jnp.stack(trial_batch.stimuli, axis=0)` directly produces the `(N, K, d)` array.

**2. Fix `SobolPlacement`**

Remove the hardcoded K=2 assumption and zero vector. `SobolPlacement.propose` should return single stimulus coordinates; the experiment loop assembles them into trial groups. If a fixed anchor is needed, it should be an explicit `anchor_stimulus: np.ndarray` parameter 

## Acceptance criteria

- [ ]  `TrialBatch.stimuli` is `list[np.ndarray]` each `(K, d)`
- [ ]  `ResponseData.add_batch` consumes a `TrialBatch` without manual conversion
